runtime/compiler/aarch64/codegen/ARM64JNILinkage.cpp

```diff
@@ -21,10 +21,240 @@
+   TR_UNIMPLEMENTED();
+   }
+
+void J9::ARM64::JNILinkage::restoreJNICallOutFrame(TR::Node *callNode, bool tearDownJNIFrame, TR::Register *vmThreadReg, TR::Register *javaStackReg)
+   {
+   TR_UNIMPLEMENTED();
+   }
+
+size_t J9::ARM64::JNILinkage::buildJNIArgs(TR::Node *callNode, TR::RegisterDependencyConditions *deps, bool passThread, bool passReceiver, bool killNonVolatileGPRs)
    {
    TR_UNIMPLEMENTED();
+   return 0;
+   }
+
+TR::Register *J9::ARM64::JNILinkage::getReturnRegisterFromDeps(TR::Node *callNode, TR::RegisterDependencyConditions *deps)
+   {
+   TR_UNIMPLEMENTED();
+   return NULL;
+   }
+
+TR::Register *J9::ARM64::JNILinkage::pushJNIReferenceArg(TR::Node *child)
+   {
+   TR_UNIMPLEMENTED();
+   return NULL;
+   }
+
+void J9::ARM64::JNILinkage::adjustReturnValue(TR::Node *callNode, bool wrapRefs, TR::Register *returnRegister)
+   {
+   TR_UNIMPLEMENTED();
+   }
+
+void J9::ARM64::JNILinkage::checkForJNIExceptions(TR::Node *callNode, TR::Register *vmThreadReg)
+   {
+   TR_UNIMPLEMENTED();
+   }
+
+TR::Instruction *J9::ARM64::JNILinkage::generateMethodDispatch(TR::Node *callNode, bool isJNIGCPoint,
+                                                               TR::RegisterDependencyConditions *deps, uintptrj_t targetAddress)
+   {
+   TR_UNIMPLEMENTED();
+   return NULL;
+   }
+
+TR::Register *J9::ARM64::JNILinkage::buildDirectDispatch(TR::Node *callNode)
+   {
+   TR::LabelSymbol *returnLabel = generateLabelSymbol(cg());
+   TR::SymbolReference *callSymRef = callNode->getSymbolReference();
+   TR::MethodSymbol *callSymbol = callSymRef->getSymbol()->castToMethodSymbol();
+   TR::ResolvedMethodSymbol *resolvedMethodSymbol = callNode->getSymbol()->castToResolvedMethodSymbol();
+   TR_ResolvedMethod *resolvedMethod = resolvedMethodSymbol->getResolvedMethod();
+   uintptrj_t targetAddress = reinterpret_cast<uintptrj_t>(resolvedMethod->startAddressForJNIMethod(comp()));
+   TR_J9VMBase *fej9 = reinterpret_cast<TR_J9VMBase *>(fe());
+
+   bool dropVMAccess = !fej9->jniRetainVMAccess(resolvedMethod);
+   bool isJNIGCPoint = !fej9->jniNoGCPoint(resolvedMethod);
+   bool killNonVolatileGPRs = isJNIGCPoint;
+   bool checkExceptions = !fej9->jniNoExceptionsThrown(resolvedMethod);
+   bool createJNIFrame = !fej9->jniNoNativeMethodFrame(resolvedMethod);
+   bool tearDownJNIFrame = !fej9->jniNoSpecialTeardown(resolvedMethod);
+   bool wrapRefs = !fej9->jniDoNotWrapObjects(resolvedMethod);
+   bool passReceiver = !fej9->jniDoNotPassReceiver(resolvedMethod);
+   bool passThread = !fej9->jniDoNotPassThread(resolvedMethod);
+
+   if (resolvedMethodSymbol->canDirectNativeCall())
+      {
+      dropVMAccess = false;
+      killNonVolatileGPRs = false;
+      isJNIGCPoint = false;
+      checkExceptions = false;
+      createJNIFrame = false;
+      tearDownJNIFrame = false;
+      }
+   else if (resolvedMethodSymbol->isPureFunction())
+      {
+      dropVMAccess = false;
+      isJNIGCPoint = false;
+      checkExceptions = false;
+      }
```

Comment:
This part is basically from x. As we do not support GPU, the code path checking `canDirectNativeCall` and `isPureFunction` is always alive.
https://github.com/eclipse/openj9/blob/e75ef990ca678cf7cf182a4e0a8c0e1b7d7af5a6/runtime/compiler/x/amd64/codegen/AMD64JNILinkage.cpp#L1349-L1366